src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,371 @@
+      setTipoEmp("No aplica");
+    }
+  };
+
+  const validarUsuario = async (mail,password,name,lastname,curp,birth_date,gender,state,town,neighborhood,program,tags,emprendedor,aliado) => {
+    try {
+      const {
+        data: { data: {  } = {} },
+      } = await validateSignUp(mail,password,name,lastname,curp,birth_date,gender,state,town,neighborhood,program,tags,emprendedor,aliado);
+      toast.success("Usuario creado exitosamente", {
+        position: "top-center",
+        autoClose: 4000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+        theme: "light",
+      });
+      
+      navigate("/login");
+    } catch (e) {
+      toast.error("Error, verifiques sus datos", {
+        position: "top-center",
+        autoClose: 4000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+        theme: "light",
+      });
+    }
+  };
+
+  const handleOnSumit = async(e) => {
+    e.preventDefault();
+    let target = e.target;
+      let datos = {
+        mail: target.mail.value,
+        password: target.password.value,
+        name: target.name.value,
+        lastname: target.lastname.value,
+        curp: target.curp.value,
+        birth_date: target.birth_date.value,
+        gender: gender,
+        state: target.state.value,
+        town: target.town.value,
+        neighborhood: target.neighborhood.value,
+        program: target.program.value,
+        tags: tag,
+        emprendedor: tipoEmp,
+        aliado: tipoAli
+      }
+      const validarCampos = await schema.isValid(datos);
+
+      if(target.password.value.length > 0 && target.password.value !== target.password2.value){
+        toast.error("Las contraseñas no coinciden", {
+          position: "top-center",
+          autoClose: 4000,
+          hideProgressBar: false,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: true,
+          progress: undefined,
+          theme: "light"
+        });
+      }
+      else if(target.password.value.length < 8){
+        toast.error("Ingrese una contraseña mayor a 8 caracteres", {
+          position: "top-center",
+          autoClose: 4000,
+          hideProgressBar: false,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: true,
+          progress: undefined,
+          theme: "light"
+        });
+      }
+      else if (validarCampos === true) {
+        validarUsuario(
+          datos.mail,
+          datos.password,
+          datos.name,
+          datos.lastname,
+          datos.curp,
+          datos.birth_date,
+          datos.gender,
+          datos.state,
+          datos.town,
+          datos.neighborhood,
+          datos.program,
+          datos.tags,
+          datos.emprendedor,
+          datos.aliado
+        );
```

Comment:
why you are sending each property instead of sending all object `datos`?

Reply:
Improve the code, I send an object now.